tapdb/multiverse.go

```diff
@@ -482,11 +487,19 @@ func (u *universeLeafCache) cacheLeafKeys(q universe.UniverseLeafKeysQuery,
 		}
 
 		// Store the cache in the top level cache.
-		_, _ = u.leafCache.Put(idStr, pageCache)
+		if _, err := u.leafCache.Put(idStr, pageCache); err != nil {
```

Comment:
Above, we're attempting to get `pageCache` from `u.leafCache`. And at this point, if `err != nil`, we're attempting to construct a new `pageCache` and put it in the `u.leafCache`.

I think that we should have a more specific error check than `err != nil` above. We should check for `cache.ErrElementNotFound` when first attempting to get `pageCache` from `u.leafCache`. If the error only relates to a missing element, then we can legitimately try to put the missing element in `u.leafCache`. But if the error is anything else, then I don't think we should try to put into `u.leafCache` at this point.